Hap score - Example Notebook

AishaDarga opened the pull request:
## Why are these changes needed?


## Related issue number (if any).

AishaDarga01 committed:
Hap Example Notebook

Signed-off-by: Aisha Mohammed Farooq Darga <aisdarg1@in.ibm.com>

AishaDarga01 committed:
Update hap_local_python.py

Signed-off-by: Aisha Mohammed Farooq Darga <aisdarg1@in.ibm.com>

AishaDarga committed:
Update hap_local_python.py

Signed-off-by: Aisha Mohammed Farooq Darga <aisdarg1@in.ibm.com>

shahrokhDaijavad commented:
@ian-cho This is another variation of a Notebook that uses your hap transform. Can you please review and provide comments?

ian-cho commented:
@shahrokhDaijavad As long as this modification does not affect other modules in the hap/python directory or make them incompatible with each other, it is fine with me.

shahrokhDaijavad commented:
Thanks, @ian-cho. So, just to confirm, you are ok with the changes made in hap_config.py and hap_local_python.py files, right? Any future changes to the hap transform directory need to be reviewed again.

ian-cho commented:
@shahrokhDaijavad Thanks for asking. If the changes affect other modules, would you mind helping resolve them when I am not available? Previously, @touma-I and you helped a lot in ensuring module compatible with each other I remember, and it was not easy for me to ensure that.

AishaDarga01 committed:
Colab compatible

Signed-off-by: Aisha Mohammed Farooq Darga <aisdarg1@in.ibm.com>

AishaDarga01 committed:
Removed the output cells

Signed-off-by: Aisha Mohammed Farooq Darga <aisdarg1@in.ibm.com>

AishaDarga commented:
@shahrokhDaijavad @ian-cho I've updated the notebook to ensure compatibility with Google Colab and included the installation of all necessary packages directly within the notebook. Could you please review it?
Thank you!

shahrokhDaijavad commented:
@AishaDarga Have you tested this notebook? I ran into errors, running both on the local machine and on the Google Colab. 

On the local machine, I get errors about hap_params not being defined. 
<img width="764" alt="image" src="https://github.com/user-attachments/assets/2acf1ce9-e08d-4d99-b470-5ff2d21453a3">
and 
<img width="763" alt="image" src="https://github.com/user-attachments/assets/7402e904-4e07-4178-969b-e562ebfd315c">

On Google Colab, I get the following error:
<img width="846" alt="image" src="https://github.com/user-attachments/assets/fe3dc222-4cf0-409d-9bfb-9548933b7f95">

AishaDarga commented:
@shahrokhDaijavad Yes, I've tested the notebook thoroughly on my system. I'll try to re-install and test it afresh. Thank you!

AishaDarga01 committed:
Addressed review comments

Signed-off-by: Aisha Mohammed Farooq Darga <aisdarg1@in.ibm.com>

AishaDarga01 committed:
Cleared output cells

Signed-off-by: Aisha Mohammed Farooq Darga <aisdarg1@in.ibm.com>

AishaDarga01 committed:
Cleared output cells

Signed-off-by: Aisha Mohammed Farooq Darga <aisdarg1@in.ibm.com>

AishaDarga committed:
Update pip install

Signed-off-by: Aisha Mohammed Farooq Darga <aisdarg1@in.ibm.com>

AishaDarga01 committed:
Remove Google Colab Compatibility

Signed-off-by: Aisha Mohammed Farooq Darga <aisdarg1@in.ibm.com>

shahrokhDaijavad commented:
@AishaDarga I just ran your latest notebook successfully, so it works fine. However, I understand you had some discussions with @touma-I about not changing the transform code. When you make the necessary changes, I will be happy to test again.

AishaDarga01 committed:
Remove the changes made to the transform and reframe the example notebook accordingly

Signed-off-by: Aisha Mohammed Farooq Darga <aisdarg1@in.ibm.com>

AishaDarga commented:
@touma-I @shahrokhDaijavad Thank you for your feedback. I've made all necessary changes in the latest commit. This version of the code leverages the HAP transform as is, without making any changes. Could you please review it

AishaDarga01 committed:
clear output cells

Signed-off-by: Aisha Mohammed Farooq Darga <aisdarg1@in.ibm.com>

AishaDarga01 committed:
Remove unnecessary files

Signed-off-by: Aisha Mohammed Farooq Darga <aisdarg1@in.ibm.com>

AishaDarga01 committed:
Remove unnecessary files

Signed-off-by: Aisha Mohammed Farooq Darga <aisdarg1@in.ibm.com>

AishaDarga01 committed:
Fixed variable mismatch

Signed-off-by: Aisha Mohammed Farooq Darga <aisdarg1@in.ibm.com>

shahrokhDaijavad reviewed (approved):
This version, which I have tested successfully, looks good to me and the directories are also clean. Thanks, @AishaDarga.

touma-I commented on examples/notebooks/hap/input-parquet/customer-feedback.parquet:
Please consider removing this file from the PR since the notebook is regenerating it every time. I think the csv-parquet transformation followed by HAP annotation is very effective without the parquet file checked to git

touma-I reviewed (changes requested):
Please consider removing the parquet file from git since we can regenerate it from css. Thanks

touma-I commented on examples/notebooks/hap/generate_hap_score_csv.ipynb:
Please consider using the released version 0.2.2:
```
! pip install  'data-prep-toolkit[ray]==0.2.2'
! pip install  'data-prep-toolkit-transforms[all]==0.2.2'
```

touma-I commented on examples/notebooks/hap/generate_hap_score_csv.ipynb:
remove data-prep-connector since you are NOT using it. This line can be deleted:

~~! pip install data-prep-connector~~

touma-I commented on examples/notebooks/hap/generate_hap_score_csv.ipynb:
You do not need this line. Those are transitive dependencies of the transform itself and could break the notebook in the future if they are changed by new releases of the ransform.

~~! pip install nltk==3.9.1 transformers==4.38.2 torch>=2.2.2,<=2.4.1 pandas==2.2.2~~

shahrokhDaijavad replied:
Adding these is my fault, @touma-I! Since I could get the notebook to work only after pip installing requirements.txt, I thought they were needed, but it turns out my problem was something else! In any case, what you are saying is that these dependencies are taken care of by the pip install of the transform itself and should not be done here.

AishaDarga01 committed:
Removed parquet folder and managed dependencies

Signed-off-by: Aisha Mohammed Farooq Darga <aisdarg1@in.ibm.com>

AishaDarga commented:
@touma-I @shahrokhDaijavad I've made all necessary changes and addressed the comments given above. I've tested the latest changes and it works fine for me. Could please review it?
<img width="683" alt="image" src="https://github.com/user-attachments/assets/49e7aee1-787b-467e-b25c-0171bb597b8c">

shahrokhDaijavad reviewed (approved):
Nice job, @AishaDarga. 
@touma-I I have tested the latest version successfully with all the changes we had asked. I recommend merging this.
